web/app/js/components/util/TapUtils.jsx

```diff
@@ -352,14 +352,19 @@ export const srcDstColumn = (d, resourceType, ResourceLink) => {
 
 export const tapLink = (d, resourceType, PrefixedLink) => {
   let namespace = d.sourceLabels.namespace;
-  let resource = "";
+  let resource;
```

Comment:
could we set namespace/resource to "" here instead, so we don't have to deal with the subsequent cases?

```suggestion
  let namespace = d.sourceLabels.namespace || "";
  let resource = "";
```

That also kind of got me thinking, that instead of forcing these to be "" to get around the fact that namespace/resource are required by TapLink, that we could instead make those prop types be not required, but have a default value of "".

Another random thought I had was maybe we should instead of doing that, add a `disabled` prop that would control whether the tap icon was greyed out. Just trying to think of whatever is clearest, it's not that important though :)

Reply:
I like the idea of `disable`, the intention would be clearer :+1: